File: src/main/java/model/json/adapter/CleanAdapter.java
package model.json.adapter;

import com.squareup.moshi.FromJson;
import com.squareup.moshi.ToJson;

public class CleanAdapter {
    public @ToJson String toJson(@Clean String data) {
        return data;
    }

    public @FromJson @Clean String fromJson(String json) {
        return json.replaceAll("[\\\\/:*?\"<>|]", "").replaceAll(" +$", "");
    }
}

File: src/main/java/model/json/adapter/CleanAdapter.kt
package model.json.adapter

import com.squareup.moshi.FromJson
import com.squareup.moshi.ToJson

class CleanAdapter {

    @ToJson
    fun toJson(@Clean data: String): String = data

    @FromJson
    @Clean
    fun fromJson(json: String): String {
        return json.replace("[\\\\/:*?\"<>|]".toRegex(), "")
                .replace(" +$".toRegex(), "")
    }

}
